doc/modules/cross_validation.rst

```diff
@@ -467,44 +379,181 @@ Here is a usage example::
 validation that allows a finer control on the number of iterations and
 the proportion of samples on each side of the train / test split.
 
+Cross-validation iterators with stratification based on class labels.
+=====================================================================
+
+For classification tasks, in some datasets the number of samples per class
+varies based on the class label. They are referred to commonly as datasets
+with imbalanced classes. In such cases, stratified sampling may be done to
+split the data in cross-validation.
 
-Label-Shuffle-Split
+The :class:`StratifiedKFold` and :class:`StratifiedShuffleSplit` do
+stratified cross-validation, that is to ensure that each training and test set
+have the same class frequencies as the overall data (up to rounding).
+class is same as far as possible.
+
+Stratified k-fold
+-----------------
+
+:class:`StratifiedKFold` is a variation of *k-fold* which returns *stratified*
+folds: each set contains approximately the same percentage of samples of each
+target class as the complete set.
+
+Example of stratified 3-fold cross-validation on a dataset with 10 samples from
+two slightly unbalanced classes::
+
+  >>> from sklearn.model_selection import StratifiedKFold
+
+  >>> X = np.ones(10)
+  >>> y = [0, 0, 0, 0, 1, 1, 1, 1, 1, 1]
+  >>> skf = StratifiedKFold(n_splits=3)
+  >>> for train, test in skf.split(X, y):
+  ...     print("%s %s" % (train, test))
+  [2 3 6 7 8 9] [0 1 4 5]
+  [0 1 3 4 5 8 9] [2 6 7]
+  [0 1 2 4 5 6 7] [3 8 9]
+
+Stratified Shuffle Split
+------------------------
+
+:class:`StratifiedShuffleSplit` is a variation of *ShuffleSplit*, which returns
+stratified splits, *i.e* which creates splits by preserving the same
+percentage for each target class as in the complete set.
+
+.. _group_cv
+
+Cross-validation iterators for grouped data.
+============================================
+
+Data is non-i.i.d. when all the samples are not drawn from an identical
+distribution. A particular case of non-i.i.d. data is data in which samples
+are grouped. The samples belonging to a particular group are identically
+distributed within their group.
```

Comment:
The problem is more that samples from within the group can be dependent on one another which breaks the iid assumption.

Group CV make sure that samples from the validation fold are independent from the samples in the training fold.